paasta_tools/contrib/check_manual_oapi_changes.sh

```diff
@@ -1,18 +1,15 @@
 #!/bin/bash
 set -uo pipefail
 
-if ! [[ "${PAASTA_SYSTEM_CONFIG_DIR:-}" =~ ^.*/general_itests/fake_etc_paasta ]] ; then
-    # Do not run during general_itests
-    diff_names=$(git diff origin/master --name-only)
-    schema_touched=$(echo "$diff_names" | grep 'paasta_tools/api/api_docs/oapi.yaml')
-    api_touched=$(echo "$diff_names" | grep 'paasta_tools/paastaapi')
+diff_names=$(git diff origin/master --name-only)
+schema_touched=$(echo "$diff_names" | grep 'paasta_tools/api/api_docs/oapi.yaml')
+api_touched=$(echo "$diff_names" | grep 'paasta_tools/paastaapi')
 
-    if [ ! -z "$api_touched" -a -z "$schema_touched" ]; then
-        echo "paasta_tools/paastaapi must not be modified manually" >&2
-        echo "Please revert your changes:" >&2
-        echo -e "$api_touched" >&2
-        exit 1
-    fi
+if [ ! -z "$api_touched" -a -z "$schema_touched" ]; then
+    echo "paasta_tools/paastaapi must not be modified manually" >&2
+    echo "Please revert your changes:" >&2
+    echo -e "$api_touched" >&2
+    exit 1
```

Comment:
It's somewhat counterintuitive to me that this case would only ever get hit during `--all-files` since the pre-commit only be invoked if $schema_touched is non-empty, so I might erroneously think we should delete this case later.

Maybe worth a quick comment?

Reply:
i'm not the greatest at bash, but from my testing + how i read this: we're hitting this if the api is touched and the schema isn't (and then below we're checking if the schema was changed and the api wasn't) - but i'm probably wrong :p

Reply:
Exactly, and so this means we're only ever going to hit this case when we run with `--all-files`, because we restrict this hook from being triggered _only_ when oapi.yaml was changed normally.  That's the part that is slightly counterintuitive to me.

Reply:
ooooh, I see what you're saying

Reply:
added a comment, lemme know if it looks good

Reply:
Perfect thank you!